backend/src/test/java/org/eclipse/tractusx/semantics/registry/JwtTokenFactory.java

```diff
@@ -12,149 +12,166 @@
-        }
-
-        public RequestPostProcessor withoutRoles(){
-            Jwt jwt = Jwt.withTokenValue("token")
-                    .header("alg", "none")
-                    .claim("sub", "user")
-                    .claim("resource_access", Map.of(publicClientId, new HashMap<String, String>()))
-                    .build();
-            Collection<GrantedAuthority> authorities = Collections.emptyList();
-            return authentication(new JwtAuthenticationToken(jwt, authorities));
-        }
-
-        private RequestPostProcessor authenticationWithRoles(String tenantId, List<String> roles){
-            Jwt jwt = Jwt.withTokenValue("token")
-                    .header("alg", "none")
-                    .claim("sub", "user")
-                    .claim("resource_access", Map.of(publicClientId, Map.of("roles", toJsonArray(roles) )))
-                    .build();
-            Collection<GrantedAuthority> authorities = Collections.emptyList();
-            return authentication(new JwtAuthenticationToken(jwt, authorities));
-        }
-
-        private static JSONArray toJsonArray(List<String> elements){
-            JSONArray jsonArray = new JSONArray();
-            for (String element : elements){
-                jsonArray.appendElement(element);
-            }
-            return jsonArray;
-        }
-    }
+public class JwtTokenFactory
+{
+
+	private static final String TENANT_ONE = "TENANT_ONE";
+	private static final String TENANT_TWO = "TENANT_TWO";
+	private static final String TENANT_THREE = "TENANT_THREE";
+
+	private final Tenant tenantOne;
+	private final Tenant tenantTwo;
+	private final Tenant tenantThree;
+
+	public JwtTokenFactory(final String publicClientId)
+	{
+		this.tenantOne = new Tenant(publicClientId, JwtTokenFactory.TENANT_ONE);
+		this.tenantTwo = new Tenant(publicClientId, JwtTokenFactory.TENANT_TWO);
+		this.tenantThree = new Tenant(publicClientId, JwtTokenFactory.TENANT_THREE);
+	}
+
+	public RequestPostProcessor allRoles()
+	{
+		return this.tenantOne.allRoles();
+	}
+
+	public RequestPostProcessor readTwin()
+	{
+		return this.tenantOne.readTwin();
+	}
+
+	public RequestPostProcessor addTwin()
+	{
+		return this.tenantOne.addTwin();
+	}
+
+	public RequestPostProcessor updateTwin()
+	{
+		return this.tenantOne.updateTwin();
+	}
+
+	public RequestPostProcessor deleteTwin()
+	{
+		return this.tenantOne.deleteTwin();
+	}
+
+	public RequestPostProcessor withoutResourceAccess()
+	{
+		return this.tenantOne.withoutResourceAccess();
+	}
+
+	public RequestPostProcessor withoutRoles()
+	{
+		return this.tenantOne.withoutRoles();
+	}
+
+	public Tenant tenantOne()
+	{
+		return this.tenantOne;
+	}
+
+	public Tenant tenantTwo()
+	{
+		return this.tenantTwo;
+	}
+
+	public Tenant tenantThree()
+	{
+		return this.tenantThree;
+	}
+
+	@Value
+	public static class Tenant
+	{
+		String publicClientId;
+		String tenantId;
+
+		public RequestPostProcessor allRoles()
```

Comment:
Could you please rebase and add the new roles which were added since the baseline.